src/main.cpp

```diff
@@ -1547,6 +1549,21 @@ bool CheckTransactionWithoutProofVerification(uint32_t tiptime,const CTransactio
                 if ( komodo_isnotaryvout(destaddr,tiptime) == 0 )
                 {
                     invalid_private_taddr = 1;
+                    int32_t current_season = getacseason(tiptime);
+                    if ( current_season > 5 && txout.scriptPubKey.IsPayToScriptHash() ) {
+                        if (out_index == tx.vout.size()-1 ) {
```

Comment:
Adding a check for 2 exactly two outputs

Reply:
My offer is don't do these checks inside vouts loop at all. May be better to leave loop ` BOOST_FOREACH(const CTxOut& txout, tx.vout)` untouched, but do all checks here:
```
if ( ASSETCHAINS_PRIVATE != 0 && invalid_private_taddr != 0 && tx.vShieldedSpend.empty() == 0 )
    {
        // here we will check that tx.vout.size() == 2, that tx.vout[0].scriptPubKey.IsPayToScriptHash() and
        // that tx.vout[0].scriptPubKey.IsRedeemScriptReveal(tx.vout[1].scriptPubKey), if not -> return a error
        return state.DoS(100, error("CheckTransaction(): this is a private chain, no sapling -> taddr"),
                         REJECT_INVALID, "bad-txns-acprivate-chain");
    }
```
Probably it will look more logic and clean.

Reply:
```
    if ( ASSETCHAINS_PRIVATE != 0 && invalid_private_taddr != 0 && tx.vShieldedSpend.empty() == 0 )
    {
        if ( !( current_season > 5
                tx.vin.size() == 0 &&
                tx.vout.size() == 2 &&
                tx.vout[0].IsPayToScriptHash() &&
                tx.vout[0].scriptPubKey.IsRedeemScriptReveal(tx.vout[1].scriptPubKey) ))
                    return state.DoS(100, error("CheckTransaction(): this is a private chain, no sapling -> taddr"),
                                     REJECT_INVALID, "bad-txns-acprivate-chain");
    }
    ```
    
Is this agreeable? Thanks, I do like that it gets rid of that iter. Definitely seemed hacky. I am addressing the rest of the feedback and I will loop back to this. Possible I missed something in the above snippet.

Reply:
Rewrite this part, resolving this thread. Feel free to reopen regarding latest changes